pkg/handler/enqueue_mapped.go

```diff
@@ -73,8 +73,13 @@ func (e *enqueueRequestsFromMapFunc) Generic(evt event.GenericEvent, q workqueue
 }
 
 func (e *enqueueRequestsFromMapFunc) mapAndEnqueue(q workqueue.RateLimitingInterface, object client.Object) {
+	reqs := map[reconcile.Request]empty{}
```

Comment:
I'm afraid this deduplication is not global enough when `mapAndEnqueue` is called several time in a single update, such as `enqueueRequestsFromMapFunc.Update`.

Reply:
Fair, I should make it work more like _owner.

Reply:
If `EnqueueRequestForOwner` had a constructor, it could share by embedding `enqueueRequestsFromMapFunc` and set `toRequests = getOwnerReconcileRequest`. 😇